src/app/wizard/step/provider-settings/provider/extended/openstack/default/component.ts

```diff
@@ -23,39 +23,37 @@ import {
   ViewChild,
 } from '@angular/core';
 import {FormBuilder, NG_VALIDATORS, NG_VALUE_ACCESSOR} from '@angular/forms';
-import {OpenstackCredentialsTypeService} from '@app/wizard/step/provider-settings/provider/extended/openstack/service';
 import {ClusterSpecService} from '@core/services/cluster-spec';
 import {PresetsService} from '@core/services/wizard/presets';
 import {FilteredComboboxComponent} from '@shared/components/combobox/component';
-import {OpenstackNetwork, OpenstackSecurityGroup, OpenstackSubnet} from '@shared/entity/provider/openstack';
+import {Cluster} from '@shared/entity/cluster';
+import {
+  OpenstackNetwork,
+  OpenstackSecurityGroup,
+  OpenstackSubnet,
+  OpenstackSubnetPool,
+} from '@shared/entity/provider/openstack';
 import {NodeProvider} from '@shared/model/NodeProviderConstants';
 import {BaseFormValidator} from '@shared/validators/base-form.validator';
 import _ from 'lodash';
 import {EMPTY, merge, Observable, onErrorResumeNext} from 'rxjs';
 import {catchError, debounceTime, filter, map, switchMap, takeUntil, tap} from 'rxjs/operators';
+import {OpenstackCredentialsTypeService} from '../service';
+import {IPVersion} from '../shared/types/ip-version';
+import {
+  IPv4SubnetIDState,
+  IPv6SubnetIDState,
+  IPv6SubnetPoolState,
+  NetworkState,
+  SecurityGroupState,
+} from '../shared/types/state';
 
 enum Controls {
   SecurityGroup = 'securityGroup',
   Network = 'network',
-  SubnetID = 'subnetID',
-}
-
-enum SecurityGroupState {
-  Ready = 'Security Group',
-  Loading = 'Loading...',
-  Empty = 'No Security Groups Available',
-}
-
-enum NetworkState {
-  Ready = 'Network',
-  Loading = 'Loading...',
-  Empty = 'No Networks Available',
-}
-
-enum SubnetIDState {
-  Ready = 'Subnet ID',
-  Loading = 'Loading...',
-  Empty = 'No Subnet IDs Available',
+  IPv4SubnetID = 'subnetID',
```

Comment:
```suggestion
  IPv4SubnetID = 'ipv4subnetID',  ? // IMO, it will make clear that we are setting value for `IPv4`
```
similarly in extended too.

i guess it won't effect because we are setting value from `onIPv4SubnetIDChange` which recieves value in an event

Reply:
Fixed.